src/Language/PureScript/Linter/Exhaustive.hs

```diff
@@ -235,21 +235,33 @@ missingAlternative env mn ca uncovered
 
 -- |
 -- Main exhaustivity checking function
--- Starting with the set `uncovered = { _ }` (nothing covered, one `_` for each function argument),
--- it partitions that set with the new uncovered cases, until it consumes the whole set of clauses.
--- Then, returns the uncovered set of case alternatives.
+-- If there are no cases, check that at least one argument is of an empty type.
+-- Otherwise, starting with the set `uncovered = { _ }` (nothing covered, one
+-- `_` for each function argument), it partitions that set with the new
+-- uncovered cases, until it consumes the whole set of clauses. Then, returns
+-- the uncovered set of case alternatives.
 --
 checkExhaustive
   :: forall m
    . (MonadWriter MultipleErrors m, MonadSupply m)
    => Environment
    -> ModuleName
-   -> Int
+   -> [Expr]
    -> [CaseAlternative]
    -> Expr
    -> m Expr
-checkExhaustive env mn numArgs cas expr = makeResult . first ordNub $ foldl' step ([initialize numArgs], (pure True, [])) cas
+checkExhaustive env mn args cas expr =
+  makeResult . first ordNub $
+    if null cas
```

Comment:
I feel like an empty case should not be treated differently. For example, what about this?

```
x :: Void -> a
x a = x a
x b = x b
```

That's redundant and exhaustive, but we will still say it's not exhaustive.

I think the trick has to be to consider the final set of binders after the `foldl`, and to realize that a wildcard binder does not necessarily represent unhandled values, since a data type might be empty.